src/rust/engine/ui/src/display.rs

```diff
@@ -349,17 +334,14 @@ impl EngineDisplay {
   pub fn finish(&mut self) {
     self.render();
     self.running = false;
-    let current_pos = self.get_cursor_pos();
-    let action_count = self.action_map.len() as u16;
-    self
-      .write(&format!(
-        "{park_cursor}{clear_after_cursor}{reveal_cursor}",
-        park_cursor = cursor::Goto(1, current_pos.1 - action_count),
-        clear_after_cursor = clear::AfterCursor,
-        reveal_cursor = termion::cursor::Show
-      ))
-      .expect("could not write to terminal");
-    self.stop_raw_mode().unwrap();
+    {
+      self.terminal = Console::Uninitialized; //This forces the AlternateScreen to drop, restoring the original terminal state.
+    }
+
+    println!("{}", termion::cursor::Show);
+    for log in self.logs.iter() {
```

Comment:
Hmm, we're not really concerned with `logs` growing too much, right?
As in, this only works because we can keep all the logs in memory, but I think it should be fine even for the largest runs.

Reply:
Yeah, I don't think it's a problem to keep the amount of logging pants does in memory, and if it ever is we can address it here somehow.